src/operator-typings.ts

```diff
@@ -0,0 +1,44 @@
+  <T2, TResult>(second: ObservableOrPromise<T2>, project: (v1: T, v2: T2) => TResult): Observable<TResult>;
+  <T2, TResult>(second: ObservableOrPromise<T2>, project: (v1: T, v2: T2) => TResult): Observable<TResult>;
+  <T2, TResult>(array: [ObservableOrPromise<T2>], project: (v1: T, v2: T2) => TResult): Observable<TResult>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>): Observable<[T, T2, T3]>;
+  <T2, T3>(array: [ObservableOrPromise<T2>, ObservableOrPromise<T3>]): Observable<[T, T2, T3]>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ObservableOrPromise<T2>, third: ArrayOrIterator<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(second: ArrayOrIterator<T2>, third: ObservableOrPromise<T3>, project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <T2, T3, TResult>(array: [ObservableOrPromise<T2>, ObservableOrPromise<T3>], project: (v1: T, v2: T2, v3: T3) => TResult): Observable<TResult>;
+  <TResult>(array: ObservableOrPromise<any>[], project?: Function): Observable<TResult[]>;
+  (...observables: Array<ObservableOrPromise<T>>): Observable<T[]>;
+  (...observables: Array<ArrayOrIterator<T>>): Observable<T[]>;
+  <R>(...observables: Array<ObservableOrPromise<T> | ((...values: Array<T>) => R)>): Observable<R>;
+  <R>(...observables: Array<ArrayOrIterator<T> | ((...values: Array<T>) => R)>): Observable<R>;
```

Comment:
This is a completely crazy amount of definitions for one method. Is this really necessary? It seems like we shouldn't have to do this, and that if this many definitions are required it's really an issue with TypeScript that should be reported and dealt with.